### DISCUSSION TITLE
Scanning the install tools for any vulnerabilities

### DISCUSSION THREAD

**Discussion by iZnoGod722** [actions/virtual-environments#6342] [General questions] [answered]
Hi guys,

I am planning on using the provided packer scripts to create images to be used for our own scale set agents within our company.

Regarding all of the tools that are installed, are these scanned by antivrus beforehand or afterwards for any vulnerabilities? 

And do I need to even be concerned about this at all?

**Comment by mikhailkoliada** [comment]
@alan722 Hello!

We are running microsoft defender during our deployment process so we are sure they are safe at least at anti-virus level.

**Comment by iZnoGod722** [comment]
Hi @mikhailkoliada thanks for the quick response.

When you say during the deployment process does this mean after or during the packer build step?

**Reply by mikhailkoliada** [reply]
@alan722 after the image is generated (i.e. packer build is done)

**Reply by iZnoGod722** [reply]
@mikhailkoliada  thanks for the clarification.

Since the only thing I am reusing in this repo are the packer build process. This means we'll need to do the scanning part ourselves once we attach this image to our scaleset?

Is possible to share with me how you guys are running microsoft defender during your deployment process?

Also Assuming the micrsoft defender scanning is only for the windows images?

**Comment by mikhailkoliada** [comment, marked as answer]
@alan722 We macOS and Ubuntu with defender too, but it won't work for you I presume as defender requires a license there.

for Windows we are using the following:

```
  Write-Host "[-] Use 100% of the CPU during the scan"
    Set-MpPreference -ScanAvgCPULoadFactor 100
    Write-Host "[-] Run Full Scan"
    $taskName = "FullScan"
    schtasks /create /tn $taskName /sc DAILY /st 13:00 /ru SYSTEM /rl HIGHEST /tr "'C:\Program Files\Windows Defender\MpCmdRun.exe' -Scan -ScanType 2"
    Start-ScheduledTask -TaskName $taskName
    while ((Get-ScheduledTask -TaskName $taskName).State  -ne 'Ready') 
    {
        Start-Sleep -Seconds 60
    }
    Write-Host "[-] Display scan results"
    $reportThreads = Get-MpThreatDetection | Format-List
    if ($reportThreads)
    {
        Write-Host "Windows Defender Antivirus found threads"
        $reportThreads
        exit 1
    }
   ```
   
   Yes, most likely you will have to do the scan manually.
   Because as you said you may wish to install the software we are not providing and as the software updates happen daily you may wish to have a slightly newer software in your image than github or azure devops provide (as we are deploying weekly).
